api/scpca_portal/models/dataset.py

```diff
@@ -106,6 +123,35 @@ def __str__(self):
         return f"Dataset {self.id}"
 
     # INSTANCE CREATION AND MODIFICATION
+    def save(self, *args, **kwargs):
+        """
+        In addition to the built-in object saving functionality,
+        cached attributes should be re-computed and re-assigned on each save.
+        """
+
+        # file hashes
+        self.data_hash = self.current_data_hash
+        self.metadata_hash = self.current_metadata_hash
+        self.readme_hash = self.current_readme_hash
+        self.combined_hash = self.current_combined_hash
+
+        # file items
+        self.includes_files_bulk = self.get_includes_files_bulk()
+        self.includes_files_cite_seq = self.get_includes_files_cite_seq()
+        self.includes_files_merged = self.get_includes_files_merged()
+
+        # stats property attributes
+        self.estimated_size_in_bytes = self.get_estimated_size_in_bytes()
+        self.diagnoses_summary = self.get_diagnoses_summary()
+        self.files_summary = self.get_files_summary()
+        self.project_diagnoses = self.get_project_diagnoses()
+        self.project_modality_counts = self.get_project_modality_counts()
+        self.modality_count_mismatch_projects = self.get_modality_count_mismatch_projects()
+        self.project_sample_counts = self.get_project_sample_counts()
+        self.project_titles = self.get_project_titles()
```

Comment:
Let's move these attributes outside of stats now that they are immediately on the model.

@nozomione this will impact you so heads up